homeassistant/components/camera/__init__.py

```diff
@@ -917,6 +962,10 @@ def _write_image(to_file: str, image_data: bytes) -> None:
         with open(to_file, "wb") as img_file:
             img_file.write(image_data)
 
+        annotation = Annotation(service_call, hass)
+        if annotation.is_defined:
+            annotation.annotate(to_file)
+
```

Comment:
It appears that you will get the image, save to disc, then open it again, edit it to add annotation, and save back to disc.

Is it possible to edit the image directly in memory before saving?

Reply:
There is some issue consuming the raw byte stream without knowing the image size in advance, I was not able to pull that but might be missing something obvious,

Reply:
You can consume it to a BytesIO object rather than write to disc.

I needed to solve the same issue here:

https://github.com/home-assistant/core/blob/540d959dd286a2531c2dd8e666dbc93afa3be8c0/homeassistant/components/generic/config_flow.py#L132

It just uses a ram buffer instead of file system storage.